File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-flutter/composed/lib/src/device_module.dart
import 'package:backend_sdk/backend_sdk.dart';

import 'context.dart';
import 'types.dart';

class CrawChatDevicesModule {
  final CrawChatSdkContext context;

  CrawChatDevicesModule(this.context);

  Future<RegisteredDeviceView?> register(RegisterDeviceRequest body) {
    return context.backendClient.device.register(body);
  }

  Future<DeviceSyncFeedResponse?> getSyncFeed(
    String deviceId, [
    CrawChatQueryParams? params,
  ]) {
    return context.backendClient.device.getDeviceSyncFeed(deviceId, params);
  }
}

File: sdks/sdkwork-craw-chat-sdk/sdkwork-craw-chat-sdk-typescript/composed/src/device-module.ts
import type {
  DeviceSyncFeedResponse,
  QueryParams,
  RegisterDeviceRequest,
  RegisteredDeviceView,
} from './types.js';
import type { CrawChatSdkContext } from './sdk-context.js';

export class CrawChatDevicesModule {
  constructor(private readonly context: CrawChatSdkContext) {}

  register(body: RegisterDeviceRequest): Promise<RegisteredDeviceView> {
    return this.context.backendClient.device.register(body);
  }

  getSyncFeed(
    deviceId: string | number,
    params?: QueryParams,
  ): Promise<DeviceSyncFeedResponse> {
    return this.context.backendClient.device.getDeviceSyncFeed(deviceId, params);
  }
}
